components/eamxx/src/physics/rrtmgp/eamxx_rrtmgp_process_interface.cpp

```diff
@@ -603,6 +615,9 @@ void RRTMGPRadiation::run_impl (const double dt) {
         });
         Kokkos::fence();
       } else {
+        if (m_co2_ramp) {
+          m_co2vmr = m_co2_begin_val + (m_co2_begin_val - m_co2_end_val) * (ts.get_year() - m_co2_begin_year);
```

Comment:
What's wrong with what you are doing here?

Reply:
Also, I think your linear ramp needs to be adjusted. I think you need to also have "end_year", so that you can do
```c++
auto slope = (m_co2_beg_val - m_co2_end_val) / (m_co2_end_year - m_co2_beg_year);
m_co2vmr = m_co2_beg_val + slope * (ts.get_year() - m_co2_beg_year);
```
In addition to "beg_val" and "beg_year" you may just ask for "slope" as input param, rather than "end_val" and "end_year".